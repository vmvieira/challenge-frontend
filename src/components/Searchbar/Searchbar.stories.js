import React from 'react';
import Searchbar from './Searchbar';

export default {
  title: 'Searchbar',
  component: Searchbar,
};

export const TheSearchbar = () => <Searchbar />;
