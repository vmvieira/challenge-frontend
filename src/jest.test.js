import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Searchbar from './components/Searchbar/Searchbar';
import { render } from '@testing-library/react';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Searchbar />, div);
});

it('renders correctly', () => {
  const { queryByPlaceholderText } = render(<Searchbar />);
  expect(queryByPlaceholderText('Search')).toBeTruthy();
});

it('matches snapshot', () => {
  const tree = renderer.create(<Searchbar />).toJSON();
  expect(tree).toMatchSnapshot();
});
