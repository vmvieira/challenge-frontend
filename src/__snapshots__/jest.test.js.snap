// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdfBwQ fDQblv"
>
  <div
    className="searchPositioning"
  >
    <label
      htmlFor="searchbar"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": "#828282",
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10.442 10.442a1 1 0 011.415 0l3.85 3.85a1 1 0 01-1.414 1.415l-3.85-3.85a1 1 0 010-1.415z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M6.5 12a5.5 5.5 0 100-11 5.5 5.5 0 000 11zM13 6.5a6.5 6.5 0 11-13 0 6.5 6.5 0 0113 0z"
          fillRule="evenodd"
        />
      </svg>
    </label>
    <input
      id="searchbar"
      placeholder="Search"
    />
  </div>
</div>
`;
